include/jemalloc/internal/rtree_inlines.h

```diff
@@ -130,21 +130,22 @@ rtree_read(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx, uintptr_t key,
 JEMALLOC_INLINE rtree_elm_t *
 rtree_elm_acquire(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx,
     uintptr_t key, bool dependent, bool init_missing) {
-	rtree_elm_t *elm;
-
-	elm = rtree_elm_lookup(tsdn, rtree, rtree_ctx, key, dependent,
-	    init_missing);
+	rtree_elm_t *elm = rtree_elm_lookup(tsdn, rtree, rtree_ctx, key,
+	    dependent, init_missing);
 	if (!dependent && elm == NULL) {
 		return NULL;
 	}
 
-	extent_t *extent;
-	void *s;
-	do {
-		extent = rtree_elm_read(elm, false);
+	spin_t spinner = SPIN_INITIALIZER;
+	while (true) {
+		extent_t *extent = rtree_elm_read(elm, false);
 		/* The least significant bit serves as a lock. */
-		s = (void *)((uintptr_t)extent | (uintptr_t)0x1);
-	} while (atomic_cas_p(&elm->pun, (void *)extent, s));
+		void *s = (void *)((uintptr_t)extent | (uintptr_t)0x1);
+		if (!atomic_cas_p(&elm->pun, (void *)extent, s)) {
+			break;
+		}
+		spin_adaptive(&spinner);
```

Comment:
Okay, how about I modify [```spin_adaptive()```](https://github.com/jemalloc/jemalloc/blob/3bd6d8e41d524737845949c44bd4961b0930965c/include/jemalloc/internal/spin_inlines.h#L16) to call ```sched_yield()``` instead of spinning once the iteration count reaches ~32?